src/broadcast.jl

```diff
@@ -0,0 +1,29 @@
+Base.Broadcast._containertype(::Type{<:DataValue}) = DataValue
+
+Base.Broadcast.promote_containertype(::Type{Any}, ::Type{DataValue}) = DataValue
+Base.Broadcast.promote_containertype(::Type{DataValue}, ::Type{Any}) = DataValue
+Base.Broadcast.promote_containertype(::Type{Tuple}, ::Type{DataValue}) = Tuple
+Base.Broadcast.promote_containertype(::Type{DataValue}, ::Type{Tuple}) = Tuple
+
+Base.Broadcast.broadcast_indices(::Type{DataValue}, A) = ()
+
+Base.Broadcast._unsafe_get_eltype(x::DataValue) = eltype(x)
+
+Base.Broadcast._broadcast_getindex_eltype(::Type{DataValue}, T::Type) = Type{T}
+Base.Broadcast._broadcast_getindex_eltype(::Type{DataValue}, A) = typeof(A)
+
+Base.@propagate_inbounds Base.Broadcast._broadcast_getindex(::Type{DataValue}, A, I) = A
+
+@inline function Base.Broadcast.broadcast_c(f, ::Type{DataValue}, a...)
+    nonnull = all(Base.hasvalue, a)
```

Comment:
So ideally I'd like mixed cases to return a ``DataValue`` in all cases. With my latest push I'm pretty close, the following all work:
````julia
julia> DataValue(3) .+ Nullable()
DataValue{Union{}}()

julia> DataValue() .+ Nullable(4)
DataValue{Union{}}()

julia> DataValue() .+ Nullable()
DataValue{Union{}}()
````
The thing that doesn't work is this:
````julia
julia> DataValue(3) .+ Nullable(2)
ERROR: MethodError: no method matching +(::Int64, ::Nullable{Int64})
Closest candidates are:
  +(::Any, ::Any, ::Any, ::Any...) at operators.jl:424
  +(::T<:Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}, ::T<:Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}) where T<:Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8} at int.jl:32
  +(::Real, ::Complex{Bool}) at complex.jl:238
  ...
Stacktrace:
 [1] broadcast_c at C:\Users\anthoff\.julia\v0.6\DataValues\src\broadcast.jl:23 [inlined]
 [2] broadcast(::Function, ::DataValues.DataValue{Int64}, ::Nullable{Int64}) at .\broadcast.jl:434
````
If you have any suggestion how to make that case also work and return a ``DataValue``, it would be great!

I'm doing all of this because I need a type that behaves like what I proposed in JuliaLang/julia#21376 for [Query.jl](https://github.com/davidanthoff/Query.jl), and I don't want to wait until this is sorted out in base...